data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,150 @@
+    @Assisted private val workerParameters: WorkerParameters,
+    private val taskService: TaskService,
+    private val taskDao: TaskDao,
+    private val reminderService: ReminderService,
+    private val reminderDao: ReminderDao,
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+
+        /** TODO is this the correct place to have this check and to retry or fail */
+        if(runAttemptCount > RETRY_COUNT) {
+            Result.failure()
+        }
+        else {
+            Result.retry()
+        }
+
+        val createdTasks: Deferred<List<TaskEntity>>
+        val updatedTasks: Deferred<List<TaskEntity>>
+        val createdReminders: Deferred<List<ReminderEntity>>
+        val updatedReminders: Deferred<List<ReminderEntity>>
+
+        /** Get all the created, updated tasks and reminders
+         *  TODO As these will run sequentially - is it better to use supervisor scope and run them in parallel? */
+        supervisorScope {
+            createdTasks = async {
+                val taskIds = taskDao.getAllTasksBySyncType(SyncAgendaType.CREATE)
+
+                taskIds.map { id ->
+                    taskDao.getTaskById(id)
+                }
+            }
+
+            updatedTasks = async {
+                val taskIds = taskDao.getAllTasksBySyncType(SyncAgendaType.UPDATE)
+
+                taskIds.map { id ->
+                    taskDao.getTaskById(id)
+                }
+            }
+
+            createdReminders = async {
+                val reminderIds = reminderDao.getAllRemindersBySyncType(SyncAgendaType.CREATE)
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+
+            updatedReminders = async {
+                val reminderIds = reminderDao.getAllRemindersBySyncType(SyncAgendaType.UPDATE)
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+        }
+
+        /** TODO I am using supervisor scope to launch so if one fails the other will still run
+         *  However, will the worker return failure if a single one fails even though some failed?
+         *  My checkResult will run everything in this block */
+        val responseState = checkResult {
+            supervisorScope {
+                val createTaskJob = launch {
+                    createdTasks.await().map { taskEntity ->
+                        /* Create the task on remote */
+                        taskService.createTask(taskEntity.toTaskDto())
+                        /* If success will run this delete - else if loop next next one */
+                        taskDao.deleteSyncTaskById(taskEntity.id)
+                    }
+                }
+
+                val updateTaskJob = launch {
+                    updatedTasks.await().forEach { taskEntity ->
+                        taskService.updateTask(taskEntity.toTaskDto())
```

Comment:
You should only delete it, if the "updateTask" was successful

Reply:
@MKiperszmid Updated with example in the above comment.